### DISCUSSION TITLE
Swift Interop with C# on Windows

### DISCUSSION THREAD

**Discussion by Noemata** [dotnet/runtime#110697] [General]
Wasn't sure where the best place is to ask this question, so here goes.

Is the .Net 9 Swift interop facility usable on Windows? I would like to use the same code on both MacOS and Windows to integrate with Swift libs on both now that the Swift compiler has reached a feasible level of maturity on Windows. If so, are there any examples out there that demonstrate integration with C# and Swift interop on Windows? If not, why is Swift compiler integration not supported on Windows? Lastly, I am aware that Swift on Windows now provides interoperability with Clang/C++, so I'm not looking at that as an alternative given the advantages of the .Net 9 approach for Swift interop.

**Comment by neon-sunset** [comment]
It might be a good idea to ask on DotNetEvolution discord server since this question isn't getting traction: https://aka.ms/dotnet-discord

(I also would like to know)

**Comment by jkotas** [comment]
This was answered in https://github.com/dotnet/runtime/issues/95638#issuecomment-2541746370

**Reply by neon-sunset** [reply]
Thanks!

**Comment by Noemata** [comment]
The logic behind not supporting the Windows side of the fence is that Swift is an Apple technology that is not relevant on Windows, even though it is becoming the most important tech on Apple devices.  Apple did a really good job of supporting Swift on Windows.  Apparently, Microsoft's ethos is not to let good intentions lead to unintended consequences (this being written on a Mac, in Safari, by a dev that has been loyal to Microsoft products for thirty years until two years ago).  C'est la vie.

**Reply by jkotas** [reply]
> The logic behind not supporting the Windows side of the fence is that Swift is an Apple technology that is not relevant on Windows

I have not said that Swift is irrelevant on Windows. I have said that Swift is a lot more popular in Apple eco system, that's why have prioritized it.

> Apple did a really good job of supporting Swift on Windows

Swift on Windows happened thanks to volunteers from the larger Swift community. Apple have not done much work on it as far as I know.

**Comment by Noemata** [comment]
@jkotas, based on commits, it looks like Apple devs are helping with the Windows side of things.  Looking at the associated .Net code for supporting Swift on Apple Silicon, it doesn't look like a parallel Windows investment would amount to a big distraction.  I don't want to ruffle any feathers, but it is important to surface how much commitment Microsoft has to its own tech stack, given recent trends.  I appreciate the clarifications.  Hopefully the folks at Microsoft will appreciate the concern Windows devs have about their futures.  It's not nearly as rosy as it once was.
